### ISSUE TITLE
Docs: Typical Playwright use cases for beginners

### ISSUE THREAD

**Issue by dzmitry-varabei** [rolling-scopes/site#232]
Please update the doc: https://github.com/rolling-scopes/site/blob/main/readme/end-to-end-testing.md

Example:
-----
# End-to-End Testing with Playwright

The project ensures that each new feature, once approved, is merged into the main branch and deployed to production. However, visual bugs (missing elements, shifted images) couldn't always be quickly detected manually. To address this, end-to-end testing with Playwright was implemented.

## How It Works

Before pushing changes to GitHub, developers can run end-to-end tests with `npx playwright test`. Playwright starts a server with the modified code, takes full-page screenshots, and compares them with reference images. Discrepancies result in test failures, with comparison results saved in the `src/__tests__/visualTesting/results` folder, helping identify and fix issues.

<img src="./assets/origin-diff.png" alt="image">

### Detecting Visual Bugs

Run `npx playwright test` to detect visual discrepancies between the current state and reference images. This is useful for catching issues like missing elements or layout shifts.

### Updating Reference Images

When intentional design changes cause test failures, delete outdated reference images (e.g., `src/__tests__/visualTesting/navbar.spec.ts-snapshots`) and rerun the tests with `npx playwright test` to create new references.

### Reviewing Test Results

Launch the Playwright UI with `npx playwright test --ui` to easily review test results and debug issues.

<img src="./assets/UI.jpg" alt="image">

## Handling Intentional Changes

If intentional changes affect the site's appearance, update the reference images by deleting the old ones and rerunning the tests. The library will create new reference samples.

## User Interface for Testing

For a user-friendly way to run and check tests, use the Playwright UI with:
```sh
npx playwright test --ui
```

### Example

If you notice a discrepancy in the navbar's appearance, the test UI will highlight the differences and provide a visual comparison to help you identify and fix the issue.

## Dealing with Errors

When tests fail, review the error messages and comparison images in the `src/__tests__/visualTesting/results` folder to understand what went wrong. Common issues include layout shifts, missing elements, or unintended changes in styling.
